aalto/python.rst

```diff
@@ -9,45 +9,101 @@ the generic `python page under scientific computing docs
 </scicomp/python>`.
 
 
-The "brain" environment with mayavi
-===================================
+The "neuroimaging" environment
+==============================
 
-On the Aalto Linux workstations there exists a conda environment under
-the anaconda2 module called "brain" which contains the mayavi
-visualization tool, mne, and pysurfer. Many of the packages in this
-environment are older than those in the main "root" environment
-because mayavi is incompatible with many newer packages. Thus it's not
-possible to install mayavi to the main root environment, and a
-separate environment is needed. To use it::
+On the Aalto Linux workstations there exists a conda environment under the
+anaconda3 module called "neuroimaging" which contains an extensive collection
+of Python packages for the analysis of neuroimaging data, such as fMRI, EEG and
+MEG.
+
+To use it::
 
     $ ml purge
-    $ ml anaconda2
-    $ source activate brain
+    $ ml anaconda3
+    $ source activate neuroimaging
+
+To see the full list of packages what are installed in the environment, use::
+
+    $ conda list
+
+Some highlights include:
+
+- Basic scientific stack
+
+  - numpy
+  - scipy
+  - matplotlib
+  - pandas
+  - statsmodels
+
+- fMRI:
+
+  - nibabel
+  - nilearn
+  - nitime
+  - pysurfer
 
-If you wish to use mayavi from the ipython terminal console or from
-the qtconsole, you need to additionally set the environment variable::
+- EEG/MEG:
 
-    $ export ETS_TOOLKIT=wx
+  - mne
+  - pysurfer
 
-If you use the spyder IDE, then you need to set Tools -> Preferences
--> Ipython console -> Graphics -> Backend: Qt.
+- Machine learning:
+
+  - scikit-learn
+  - tensorflow
+  - pytorch
+
+- R:
+
+  - rpy2 (bridge between Python and R)
+  - tidyverse
+
+If you frequently use the "neuroimaging" environment, you may consider making
+it your default Python environment by adding these lines to your ``~/.zshrc``::
+
+	# Set default python environment to "neuroimaging"
+	module load anaconda3
+	export CONDA_DEFAULT_ENV="/work/modules/Ubuntu/14.04/amd64/common/anaconda3/latest/envs/neuroimaging"
+	export PATH="$CONDA_DEFAULT_ENV/bin:$PATH"
```

Comment:
This is not necessarily a good idea, actually. I was just solving a problem for a user where libreoffice crashed when opening a new document. The culprit turned out to be that he had installed miniconda3 in his home directory, and added that to the default $PATH. Then libreoffice tried to use that python, and due to some incompatibility it crashed.

Reply:
ah, yes. Good point.